python/lsst/pipe/tasks/rgb2hips/_all_sky_hips.py

```diff
@@ -0,0 +1,521 @@
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = ("AllSkyHipsTaskConnections", "AllSkyHipsTaskConfig", "AllSkyHipsTask")
+
+from PIL import Image
+import io
+import math
+import re
+import hpgeom as hpg
+import numpy as np
+import healsparse as hsp
+from dataclasses import replace
+from datetime import datetime
+from collections.abc import Iterable
+from astropy.io import fits
+
+
+from lsst.resources import ResourcePath
+from lsst.pex.config import Field, ConfigField, ChoiceField
+from lsst.sphgeom import RangeSet
+
+from lsst.pipe.base import (
+    PipelineTask,
+    PipelineTaskConfig,
+    PipelineTaskConnections,
+    Struct,
+    QuantumContext,
+    InputQuantizedConnection,
+    OutputQuantizedConnection,
+    TaskMetadata,
+)
+from lsst.pipe.base.connectionTypes import Input
+
+
+from ._properties import HipsPropertiesConfig, _write_property
+from ..healSparseMapping import _is_power_of_two
+
+
+class AllSkyHipsTaskConnections(
+    PipelineTaskConnections,
+    dimensions=tuple(),
+    defaultTemplates={"input_task_label": "generateLowOrderHips"},
+):
+    low_order_metadata = Input(
+        doc="Metadata produced by the LowOrderHipsTask",
+        name="{input_task_label}_metadata",
+        storageClass="TaskMetadata",
+        multiple=True,
+        deferLoad=True,
+        dimensions=tuple(),
+    )
+    input_hips = Input(
+        doc="Hips pixels at level 8 used to build higher orders",
+        name="rgb_picture_hips8",
+        storageClass="NumpyArray",
+        multiple=True,
+        deferLoad=True,
+        dimensions=("healpix8",),
+    )
+
+    def __init__(self, *, config: AllSkyHipsTaskConfig):
+        # Set the input dimensions to whatever the minimum order healpix
+        # to produce is.
+        self.low_order_metadata = replace(
+            self.low_order_metadata, dimensions=set((f"healpix{config.min_order}",))
+        )
+
+
+class AllSkyHipsTaskConfig(PipelineTaskConfig, pipelineConnections=AllSkyHipsTaskConnections):
+    hips_base_uri = Field[str](
+        doc="URI to HiPS base for output.",
+        optional=False,
+    )
+    color_ordering = Field[str](doc="The bands used to construct the input images", optional=False)
```

Comment:
This is a single value (`Field` not `ListField`) but the doc string says "bands".  I think this needs more documentation.